src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -96,11 +98,19 @@ public async Task SynchronizeWithServer()
                 return;
             }
 
+            if (!IsFeatureSupported())
+            {
+                logger.WriteLine(TaintResources.Synchronizer_UnsupportedSQVersion, sonarQubeService.ServerInfo.Version);
+                ClearStore();
```

Comment:
This is a 3rd prerequisite that we have and the code is starting to get a bit messy with the copy paste.

Perhaps add helper methods that log the other 2 conditions, so we could do something like
```
if (IsStandalone() || !IsFeatureSupported() || !ServerConnected())
{
   ClearStore();
   UpdateTaintIssuesUIContext(false);
   return;
}
```
_Technically for `sonarQubeService.IsConnected` we don't clear the store, but I think that's just an oversight. I don't see a reason why we shouldn't._

Reply:
I did most of that and ended up reverting it. I can't remember if there was a good reason for the `IsConnected` check not clearing the store, but there's an explicit test for it. Same for the exception handling; I couldn't remember if there was reason we didn't clear the store in that case.
Making those changes introduced several changes of behaviour that were unrelated to the bug. I'd rather keep them separate; this commit fixes the bug. Changing the other behaviours and refactoring should be separate.

Reply:
In the exception handling we do the opposite: we clear the store but don't reset the UI context. I find that odd.

In the IsConnected case, I believe I did that because I assumed that we will be connected "shortly", and that will either set the store or clear it, so it was more of an optimization. Originally it did nothing except return, and when the UI context was introduced, it suddenly did "something" but we didn't go back and made sure it did the same logic as the other cases.

Either way, I think it's worth doing the cleanup and investigating if there was a reason for the difference. This code was written less than a month ago and we're already "scared" to touch it. We should either clean it up, or understand what are the reasons for the difference and document them. That being said, I'm ok with doing that in a separate commit if you prefer.

Reply:
I don't have a problem with changing the behaviour, but it's unrelated to the bug fix so it should be done separately.